packages/base/source/helpers/assert-has.js

```diff
@@ -0,0 +1,24 @@
+//----------------------------------------------------------------------
+//
+// This source file is part of the Folktale project.
+//
+// Licensed under MIT. See LICENCE for full licence information.
+// See CONTRIBUTORS for the list of contributors to the project.
+//
+//----------------------------------------------------------------------
+
+// [jwarden 7.14.2017] NOTE: Can of worms below. I'm concerned based on how much work went into Lodash's has for example.
+// And has only checks Objects, not if it's a class and walking up the prototype tree.
+// Sooo... this assumes:
+// 1. it's an Object, not a class, nor Array
+// 2. a basic hasOwnProperty check 
+module.exports = (method, object, property, customErrorMessage = '') => {
+    if (!object || typeof property !== 'string' || object.hasOwnProperty(property) === false) {
```

Comment:
Ah, this only needs an `in` check, as matchWith doesn't care where the property is, just that it can find that in the object. So:

```js
if (!(property in object)) { throw new Error(...) }
```